scrapy/utils/python.py

```diff
@@ -254,25 +254,22 @@ def get_spec(func):
 def equal_attributes(obj1, obj2, attributes):
     """Compare two objects attributes"""
     # not attributes given return False by default
-    if not attributes:
-        return False
-
-    for attr in attributes:
-        # support callables like itemgetter
-        if callable(attr):
-            if not attr(obj1) == attr(obj2):
-                return False
+    res = False
```

Comment:
I don't think Dijkstra's style is a good think to Python; http://anthonysteele.co.uk/the-single-exit-point-law provide some arguments I agree with. IMHO using return is fine: it is shorter and easier to read. In you implementation developer have to check if `res` is changed later or not to understand what is code doing; `return False` allows a more local reasoning - less things to keep in mind.